testutils/src/main/java/com/google/android/exoplayer2/testutil/truth/SpannedSubject.java

```diff
@@ -1110,4 +1152,95 @@ public String toString() {
+  public interface TextEmphasisDescription {
+
+    /**
+     * Checks that at least one of the matched spans has the expected {@code mark} and {@code position}.
+     *
+     * @param mark The expected mark
+     * @param position The expected position of the mark
+     * @return A {@link WithSpanFlags} object for optional additional assertions on the flags.
+     */
+    AndSpanFlags withMarkAndPosition(@TextEmphasisSpan.Mark int mark,
+        @TextEmphasisSpan.Position int position);
+  }
+
+  private static final TextEmphasisDescription ALREADY_FAILED_WITH_MARK =
+      (mark, position) -> ALREADY_FAILED_AND_FLAGS;
+
+  private static Factory<TextEmphasisSubject, List<TextEmphasisSpan>> textEmphasisSubjects(Spanned actualSpanned) {
+    return (FailureMetadata metadata, List<TextEmphasisSpan> spans) ->
+        new TextEmphasisSubject(metadata, spans, actualSpanned);
+  }
+
+  private static final class TextEmphasisSubject extends Subject implements TextEmphasisDescription {
+
+    private final List<TextEmphasisSpan> actualSpans;
+    private final Spanned actualSpanned;
+
+    private TextEmphasisSubject(
+        FailureMetadata metadata, List<TextEmphasisSpan> actualSpans, Spanned actualSpanned) {
+      super(metadata, actualSpans);
+      this.actualSpans = actualSpans;
+      this.actualSpanned = actualSpanned;
+    }
+
+    @Override
+    public AndSpanFlags withMarkAndPosition(@TextEmphasisSpan.Mark int mark,
+        @TextEmphasisSpan.Position int position) {
+      List<Integer> matchingSpanFlags = new ArrayList<>();
+      List<MarkAndPosition> textEmphasisMarksAndPositions = new ArrayList<>();
+      for (TextEmphasisSpan span : actualSpans) {
+        textEmphasisMarksAndPositions.add(new MarkAndPosition(span.mark, span.position));
+        if (span.mark == mark && span.position == position) {
+          matchingSpanFlags.add(actualSpanned.getSpanFlags(span));
+        }
+      }
+      check("textEmphasisMarkAndPosition")
+          .that(textEmphasisMarksAndPositions)
+          .containsExactly(new MarkAndPosition(mark, position));
+      return check("flags").about(spanFlags()).that(matchingSpanFlags);
+    }
+
+    private static final class MarkAndPosition {
+
+      @TextEmphasisSpan.Mark
+      private final int mark;
+      @TextEmphasisSpan.Position
+      private final int position;
+
+      private MarkAndPosition(@TextEmphasisSpan.Mark int mark,
+          @TextEmphasisSpan.Position int position) {
+        this.mark = mark;
+        this.position = position;
+      }
+
+      @Override
+      public boolean equals(@Nullable Object o) {
+        if (this == o) {
+          return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+          return false;
+        }
+
+        TextEmphasisSubject.MarkAndPosition that = (TextEmphasisSubject.MarkAndPosition) o;
+        return (position == that.position) && (mark == that.mark);
+      }
+
+      @Override
+      public int hashCode() {
```

Comment:
I think we can remove this and equals() - not sure it really adds much for a Truth subject implementation

Reply:
Actually, these are required to avoid checking for instance equality using Truth.

Reply:
Ah misread this and thought it was on the subject not the inner MarkAndPosition - my bad! 🙈